qdrant_client/qdrant_fastembed.py

```diff
@@ -0,0 +1,287 @@
+        embedding_model: str = DEFAULT_EMBEDDING_MODEL,
+        **kwargs: Any,
+    ) -> List[types.ScoredPoint]:
+        """
+        Search for documents in a collection.
+        This method automatically encodes the query text using the specified embedding model.
+        If you want to use your own encoded query vector, use `search` method instead.
+
+        Args:
+            collection_name: Collection to search in
+            query_text:
+                Text to search for. This text will be encoded using the specified embedding model.
+                And then used as a query vector.
+            query_filter:
+                - Exclude vectors which doesn't fit given conditions.
+                - If `None` - search among all vectors
+            search_params: Additional search params
+            limit: How many results return
+            offset:
+                Offset of the first result to return.
+                May be used to paginate results.
+                Note: large offset values may cause performance issues.
+            with_payload:
+                - Specify which stored payload should be attached to the result.
+                - If `True` - attach all payload
+                - If `False` - do not attach any payload
+                - If List of string - include only specified fields
+                - If `PayloadSelector` - use explicit rules
+            with_vectors:
+                - If `True` - Attach stored vector to the search result.
+                - If `False` - Do not attach vector.
+                - If List of string - include only specified fields
+                - Default: `False`
+            score_threshold:
+                Define a minimal score threshold for the result.
+                If defined, less similar results will not be returned.
+                Score of the returned result might be higher or smaller than the threshold depending
+                on the Distance function used.
+                E.g. for cosine similarity only higher scores will be returned.
+            embedding_model:
+                Which embedding model to use. Defaults to "fast-all-MiniLM-L6-v2".
+                Models are automatically downloaded by `fastembed` library.
+
+        Returns:
+            List[types.ScoredPoint]: List of scored points.
+
+        """
+        embedding_model_inst = self._get_or_init_model(model_name=embedding_model)
+        embeddings = list(embedding_model_inst.encode(documents=[query_text]))
+        query_vector = embeddings[0][0]
+
+        return self.search(
+            collection_name=collection_name,
+            query_vector=query_vector,
+            query_filter=query_filter,
+            search_params=search_params,
+            limit=limit,
+            offset=offset,
+            with_payload=with_payload,
+            with_vectors=with_vectors,
+            score_threshold=score_threshold,
+            **kwargs,
+        )
```

Comment:
A major point of API improvement is a simpler Response type, this is why I created `QueryResponse` and simplified the `ScoredPoint` and discarded a lot of information. 

Would it possible to retain that here?

Reply:
Could you please elaborate how plain dict it better than a type?
Previous version converted `QueryResponse` into `dict` before returning it, in fact  `QueryResponse` was only used internally and gave no info to the user about the response structure.

If you propose to replace return type `ScoredPoint -> QueryResponse` that I can agree on